geowebcache/wms/src/test/java/org/geowebcache/service/wms/WMSTileFuserTest.java

```diff
@@ -78,23 +96,26 @@ public void testTileFuserResolution() throws Exception {
         int width = (int) bounds.getWidth() * 10;
         int height= (int) bounds.getHeight() * 10;
         GridSubset gridSubset = layer.getGridSubset(layer.getGridSubsets().iterator().next());
-        WMSTileFuser tileFuser = new WMSTileFuser(layer, gridSubset, bounds, width, height);
+        TileLayerDispatcher tld = mock(TileLayerDispatcher.class);
+        StorageBroker sb = mock(StorageBroker.class);
+        Mockito.when(tld.getTileLayer("test:layer")).thenReturn(layer);
+        WMSTileFuser tileFuser = new WMSTileFuser(tld, sb, fuserRequest(layer, gridSubset, bounds, width, height));
         tileFuser.setSecurityDispatcher(secDisp);
         tileFuser.determineSourceResolution();
         assertEquals(0.087890625, tileFuser.srcResolution, 0.087890625*0.001);
         
         // Zoomed too far out
         height = (int) bounds.getWidth() / 10;
         width = (int) bounds.getWidth() / 10;
-        tileFuser = new WMSTileFuser(layer, gridSubset, bounds, width, height);
+        tileFuser = new WMSTileFuser(tld, sb, fuserRequest(layer, gridSubset, bounds, width, height));
```

Comment:
With these changes, this no longer tests the `@Deprecated` constructor. 

If the constructor is still used anywhere in GWC, a simple test using that constructor is probably a good idea, given the amount of setup that seems to be done in it. If that constructor is not used anywhere anymore, seems fine as is.

Reply:
The constructor appeared to exist only for the sake of the tests, resulting in different behaviour between what was tested and what happens in real usage.  I considered removing the constructor entirely bu deprecated to play it safe although if there's anything using it out there it will likely fall afoul of NPEs in the security filter.

Reply:
Ok, in that case everything looks good.